pallets/asset-index/src/lib.rs

```diff
@@ -355,32 +355,21 @@ pub mod pallet {
 			location: MultiLocation,
 			amount: T::Balance,
 		) -> DispatchResultWithPostInfo {
-			let caller = T::AdminOrigin::ensure_origin(origin)?;
-			if units.is_zero() {
-				return Ok(().into());
-			}
-
-			let availability = AssetAvailability::Liquid(location);
-
-			// check whether this is a new asset and make sure locations match otherwise
-			let is_new_asset = if let Some(asset) = Assets::<T>::get(&asset_id) {
-				ensure!(asset == availability, Error::<T>::AssetAlreadyExists);
-				false
-			} else {
-				true
-			};
-
-			// transfer the caller's fund into the treasury account
-			Self::add_liquid(&caller, asset_id, units, amount)?;
-
-			// register asset if not yet known
-			if is_new_asset {
-				Assets::<T>::insert(asset_id, availability.clone());
-				Self::deposit_event(Event::AssetRegistered(asset_id, availability));
-			}
+			Self::do_add_asset(T::AdminOrigin::ensure_origin(origin)?, asset_id, units, location, amount)
+		}
 
-			Self::deposit_event(Event::AssetAdded(asset_id, units, caller, amount));
-			Ok(().into())
+		/// Add liquid asset with root origin, see `add_asset`
+		#[pallet::weight(T::WeightInfo::add_asset())]
+		pub fn force_add_asset(
+			origin: OriginFor<T>,
+			recipient: T::AccountId,
```

Comment:
can we make this the last parameter, so that the parameter of `add_asset` and `force_add_asset` are identical?